pep8 ? est-ce lint backend ?

```
* Run lint for backend
./zds/tutorialv2/publication_utils.py:9:1: F401 'os.mkdir' imported but unused
./zds/tutorialv2/api/views.py:80:1: W293 blank line contains whitespace
./zds/tutorialv2/management/commands/publication_watchdog.py:14:1: E302 expected 2 blank lines, found 1
```

--- a/scripts/define_function.sh
+++ b/scripts/define_function.sh
@@ -60,6 +60,7 @@ function zds_start_zmd {
     fi
 }
 
+
 function zds_stop_zmd {
     pm2 kill; exVal=$?
 
@@ -68,6 +69,14 @@ function zds_stop_zmd {
     fi
 }
 ## end
+
+
+function gateway {
+    if [[ $2 != 0 ]]; then
+        print_error "$1"
+        exit $2
+    fi
+}
 
 
 ## start print function

--- a/scripts/define_function.sh
+++ b/scripts/define_function.sh
@@ -73,6 +73,7 @@ function zds_stop_zmd {
 
 function gateway {
     if [[ $2 != 0 ]]; then
+        zds_fold_end
         print_error "$1"
         exit $2
     fi

--- a/scripts/travis_script.sh
+++ b/scripts/travis_script.sh
@@ -10,7 +10,9 @@ exVal=0
 # start elastic
 if [[ "$1" == "start_elasticsearch" ]] && [[ "$ZDS_TEST_JOB" == *"zds.searchv2"* ]]; then
     zds_fold_start "elasticsearch" "* Start elasticsearch as service"
-        sudo service elasticsearch start
+        sudo service elasticsearch start; exVal=$?
+
+        gateway "!! elasticsearch didn't start" $exVal
     zds_fold_end
 fi
 
@@ -18,7 +20,9 @@ fi
 # start latex
 if [[ "$1" == "start_latex" ]] && [[ "$ZDS_TEST_JOB" == *"zds.tutorialv2"* ]]; then
     zds_fold_start "latex" "* Start texhash -> latex"
-        texhash
+        texhash; exVal=$?
+
+        gateway "!! Texhash failed" $exVal
     zds_fold_end
 fi
 
@@ -28,14 +32,18 @@ if [[ "$1" == "lint_backend" ]] && [[ "$ZDS_TEST_JOB" == *"zds.gallery"* ]]; the
     zds_fold_start "lint_backend" "* Run lint for backend"
         ./scripts/no_import_zds_settings.sh \
             && flake8 \
-            && flake8 --config=zds/settings/.flake8 zds/settings
+            && flake8 --config=zds/settings/.flake8 zds/settings; exVal=$?
+
+        gateway "!! Test failed" $exVal
     zds_fold_end
 fi
 
 # test backend
 if [[ "$1" == "test_backend" ]] && [[ "$ZDS_TEST_JOB" == *"zds."* ]]; then
     zds_fold_start "test_backend" "* Run test for backend"
-        python manage.py makemigrations --dry-run --check
+        python manage.py makemigrations --dry-run --check; exVal=$?
+
+        gateway "!! Test failed" $exVal
     zds_fold_end
 fi
 
@@ -43,7 +51,9 @@ fi
 # lint frontend
 if [[ "$1" == "lint_frontend" ]] && [[ "$ZDS_TEST_JOB" == *"front"* ]]; then
     zds_fold_start "lint_frontend" "* Run lint for frontend"
-        npm run lint
+        npm run lint; exVal=$?
+
+        gateway "!! Test failed" $exVal
     zds_fold_end
 fi
 
@@ -59,7 +69,9 @@ if [[ "$1" == "coverage_backend" ]] && [[ "$ZDS_TEST_JOB" == *"zds."* ]]; then
             --keepdb \
             --settings zds.settings.ci_test \
             --exclude-tag=front \
-            ${ZDS_TEST_JOB/front/}
+            ${ZDS_TEST_JOB/front/}; exVal=$?
+
+        gateway "!! Test failed" $exVal
 
         zds_stop_zmd
 
@@ -85,7 +97,7 @@ if [[ "$1" == "selenium_test" ]]  && [[ "$ZDS_TEST_JOB" == *"selenium"* ]]; then
             test -v=2\
             --settings zds.settings.ci_test \
             --tag=front \
-            --keepdb
+            --keepdb; exVal=$?
 
         zds_stop_zmd
 
@@ -98,7 +110,8 @@ if [[ "$1" == "build_documentation" ]] && [[ "$ZDS_TEST_JOB" == *"doc"* ]]; then
     zds_fold_start "doc" "* Run SphinxBuild to build documentation"
         print_info "* Build documentation"
         if [[ "$ZDS_TEST_JOB" == *"doc"* ]]; then
-            make generate-doc
+            make generate-doc; exVal=$?
+            gateway "!! Build documentation failed" $exVal
         fi
     zds_fold_end
 fi

